### PR TITLE
Fix spacing

### PR DESCRIPTION
**Pull request by broccolini**
This pr updates the spacing to work with the correct Primer spacing scale and breakpoints.

The majority of things match exactly, but some things are intentionally slightly different, and a few things I couldn't get to match easily and didn't seem worth it.

To provide feedback, I'd suggest testing on mobile, ipad, and desktop and seeing if anything looks awry. You can compare to prod though don't get hung up on small discrepancies if it looks food and doesn't impact the user experience.

Also in this pr:
- updated metadata title from `GitHub Primer open-source projects` to `Primer open-source projects`
- change `Primer React` to `Primer Components`

### CODE REVIEW

File: src/Hero.js [hunk 1]
```diff
@@ -7,24 +7,24 @@ import IndexGrid from './IndexGrid'
 
 export default function Hero() {
   return (
-    <Box pt={[4, 2, 3]}>
-      <IndexGrid my={[3, 3, 6]} flexDirection={[`row`, `row`, `row-reverse`]} alignItems="end">
-        <FlexItem px={4} width={[10 / 12, 8 / 12, 6 / 12, 5 / 12]} mx={'auto'}>
+    <Box pt={[4, 2, 5]}>
+      <IndexGrid my={[6, 6, 12]} flexDirection={[`row`, `row`, `row-reverse`]} alignItems="end">
```
Comment (emplums) [inline src/Hero.js]: These all can be regular single quotes, surprised the linter didn't pick that up! Looks like it was like that before though so not a big deal to change

File: src/Hero.js [hunk 2]
```diff
@@ -7,24 +7,24 @@ import IndexGrid from './IndexGrid'
...
-        <FlexItem px={4} my={[4, 3, 0]} width={[1, 1, 6 / 12, 7 / 12]}>
+        <FlexItem px={5} my={[4, 3, 0]} width={[1, 1, 1, 7 / 12]}>
           <Heading color="blue.4" mb={2} fontSize={[48, 56, 84]} fontWeight="bold">
             Primer
           </Heading>
           <Text color="blue.2" fontSize={[4, 5, 5, 7]} lineHeight={1.25}>
             Resources, tooling, and design guidelines for building interfaces with GitHub’s design system
           </Text>
-          <Text is="p" color="blue.3" mt={4} className="text-mono">
+          <Text is="p" color="blue.3" mt={5} className="text-mono">
```
Comment (emplums) [inline src/Hero.js]: You can use `fontFamily='mono'` here now that we've got the full theme imported, but we could do a separate clean up PR to get rid of all these instances

**Comment by emplums** [comment]
Do we want the items in the nav to drop to two lines on iPhone size screens? 
image

**Comment by broccolini** [comment]
> Do we want the items in the nav to drop to two lines on iPhone size screens?

I wasn't going to touch that in this branch since you have edited the link styles and Rohan has removed the news link. I'll tackle that in follow up.

**Comment by emplums** [comment]
Not sure if we're supporting responsive styles on this small of a screen, butiPhone 5 sized screens, these buttons have no spacing between them (same with the buttons in the Atom section):
image

**Pull request merged**